File: script/generate-logos.py
#!/usr/bin/env python

"""
Script to generate logo variants from robot-logo.png
Generates:
- mark-16.png: 16x16 logo
- mark-32.png: 32x32 logo
- mark-48.png: 48x48 logo
- mark-64.png: 64x64 logo
- mark-96.png: 96x96 logo
- mark-128.png: 128x128 logo
- mark-128-padded.png: 128x128 logo with 96x96 content + 16px padding
- mark-256.png: 256x256 logo
"""

import os
import sys
from pathlib import Path

try:
    from PIL import Image
except ImportError:
    print("Error: Pillow (PIL) is not installed.")
    print("Please install it with: pip install Pillow")
    sys.exit(1)


def generate_logos():
    """Generate logo variants from robot-logo.png"""
    
    script_dir = Path(__file__).parent
    project_root = script_dir.parent
    assets_dir = project_root / "robotframework-recorder-assets"
    output_dir = project_root / "assets"
    source_file = assets_dir / "robot-logo.png"
    
    # Check if source file exists
    if not source_file.exists():
        print(f"❌ Error: Source file not found: {source_file}")
        sys.exit(1)
    
    print("🤖 Generating logo variants...\n")
    
    # Open the source image
    try:
        img = Image.open(source_file)
    except Exception as e:
        print(f"❌ Error opening source file: {e}")
        sys.exit(1)
    
    # Convert to RGBA if needed
    if img.mode != 'RGBA':
        img = img.convert('RGBA')
    
    # Define all sizes
    sizes = [16, 32, 48, 64, 96, 128, 256]
    
    # Generate each size
    for size in sizes:
        print(f"📐 Generating mark-{size}.png ({size}x{size})...")
        try:
            logo = Image.new('RGBA', (size, size), (255, 255, 255, 0))
            
            # Resize the source image to fit within the target size
            img_resized = img.copy()
            img_resized.thumbnail((size, size), Image.Resampling.LANCZOS)
            
            # Paste the resized image in the center
            offset = ((size - img_resized.width) // 2, (size - img_resized.height) // 2)
            logo.paste(img_resized, offset, img_resized)
            
            # Save as PNG
            logo.save(output_dir / f"mark-{size}.png", 'PNG')
            print(f"✅ mark-{size}.png created successfully")
        except Exception as e:
            print(f"❌ Error creating mark-{size}.png: {e}")
            sys.exit(1)
    
    print()
    
    # Generate mark-128-padded.png
    print("📐 Generating mark-128-padded.png (128x128 with padding)...")
    try:
        # Create a new image with transparent background (128x128)
        logo_128 = Image.new('RGBA', (128, 128), (255, 255, 255, 0))
        
        # Resize the source image to 96x96
        img_resized = img.copy()
        img_resized.thumbnail((96, 96), Image.Resampling.LANCZOS)
        
        # Paste the 96x96 image in the center (leaving 16px padding on each side)
        offset = ((128 - img_resized.width) // 2, (128 - img_resized.height) // 2)
        logo_128.paste(img_resized, offset, img_resized)
        
        # Save as PNG
        logo_128.save(output_dir / "mark-128-padded.png", 'PNG')
        print("✅ mark-128-padded.png created successfully\n")
    except Exception as e:
        print(f"❌ Error creating mark-128-padded.png: {e}")
        sys.exit(1)
    
    print("🎉 All logos generated successfully!")
    print(f"📁 Output location: {output_dir}")


if __name__ == "__main__":
    generate_logos()

File: script/generate-logos.js
#!/usr/bin/env node
 

/**
 * Script to generate logo variants from robot-logo.png
 * Generates:
 * - mark-16.png: 16x16 logo
 * - mark-32.png: 32x32 logo
 * - mark-48.png: 48x48 logo
 * - mark-64.png: 64x64 logo
 * - mark-96.png: 96x96 logo
 * - mark-128.png: 128x128 logo
 * - mark-128-padded.png: 128x128 logo with 96x96 content + 16px padding
 * - mark-256.png: 256x256 logo
 */

const fs = require('fs');
const path = require('path');

// Check if Sharp is installed (optional dependency)
/* eslint-disable import/no-unresolved */
let sharp;
try {
  sharp = require('sharp');
} catch (error) {
  console.error('Error: Sharp package is not installed.');
  console.error('Please install it first with: npm install sharp');
  process.exit(1);
}
/* eslint-enable import/no-unresolved */

const assetsDir = path.join(__dirname, '..', 'robotframework-recorder-assets');
const outputDir = path.join(__dirname, '..', 'assets');
const sourceFile = path.join(assetsDir, 'robot-logo.png');

// Check if source file exists
if (!fs.existsSync(sourceFile)) {
  console.error(`Error: Source file not found: ${sourceFile}`);
  process.exit(1);
}

console.log('🤖 Generating logo variants...\n');

const sizes = [16, 32, 48, 64, 96, 128, 256];
const promises = [];

// Generate each size
sizes.forEach((size) => {
  console.log(`📐 Generating mark-${size}.png (${size}x${size})...`);

  const promise = sharp(sourceFile)
    .resize(size, size, {
      fit: 'contain',
      background: {
        r: 255, g: 255, b: 255, alpha: 0
      }
    })
    .png()
    .toFile(path.join(outputDir, `mark-${size}.png`))
    .then(() => {
      console.log(`✅ mark-${size}.png created successfully`);
    })
    .catch((err) => {
      console.error(`❌ Error creating mark-${size}.png:`, err.message);
      process.exit(1);
    });

  promises.push(promise);
});

// Generate mark-128-padded.png after all sizes are done
Promise.all(promises)
  .then(() => {
    console.log('\n📐 Generating mark-128-padded.png (128x128 with padding)...');
    return sharp(sourceFile)
      .resize(96, 96, {
        fit: 'contain',
        background: {
          r: 255, g: 255, b: 255, alpha: 0
        }
      })
      .extend({
        top: 16,
        bottom: 16,
        left: 16,
        right: 16,
        background: {
          r: 255, g: 255, b: 255, alpha: 0
        }
      })
      .png()
      .toFile(path.join(outputDir, 'mark-128-padded.png'));
  })
  .then(() => {
    console.log('✅ mark-128-padded.png created successfully');
    console.log('\n🎉 All logos generated successfully!');
    console.log(`📁 Output location: ${outputDir}`);
  })
  .catch((err) => {
    console.error('❌ Error creating mark-128-padded.png:', err.message);
    process.exit(1);
  });
